src/bsmHandler.cpp

```diff
@@ -376,110 +375,27 @@ bool BSMHandler::process( const std::string& bsm_json ) {
     return result_ == ResultStatus::SUCCESS;
 }
 
-void BSMHandler::handlePartIIRedaction(rapidjson::Value& data) {
-    bool debug = false;
-    
-    int numMembersRedacted = 0;
-
-    // check if partII redaction is required
-    if (data.HasMember("partII") && is_active<kPartIIRedactFlag>()) {
-        if (debug) { std::cout << "partII redaction is required" << std::endl; }
-
-        // get partII data
-        rapidjson::Value& partII = data["partII"];
-
-        // for each field
-        for (std::string member : rpm.getFields()) {
-            if (debug) {
-                bool psuccess = false;
-                isMemberPresent(partII, member, psuccess);
-                std::cout << "Is the '" << member << "' member present... Before redaction? " << psuccess;
-            }
-
-            // redact field
-            bool success = false;
-            findAndRemoveAllInstancesOfMember(partII, member.c_str(), success);
-            if (success) {
-                numMembersRedacted++;
-            }
-
-            if (debug) {
-                bool psuccess = false;
-                isMemberPresent(partII, member, psuccess);
-                std::cout << "----- After redaction? " << psuccess << std::endl;
-            }
+void BSMHandler::handleGeneralRedaction(rapidjson::Document& document) {
+    if (is_active<kGeneralRedactFlag>()) {
+        for (std::string memberPath : rpm.getFields()) {
+            bool memberRedacted = false;
+            rapidjsonRedactor.redactMemberByPath(document, memberPath.c_str(), memberRedacted);
```

Comment:
We just talked about this on the phone, but I wanted to get it into the PR as well. Let's update the redactMemberByPath to return a boolean. Then we can log (at info level) if we failed to remove the member, and we can add some efficiencies into the array iterator in the method itself to break out early in the loop instead of iterating all members after finding our value.